app/src/main/java/com/monkeyteam/chimpagne/ui/event/polls/PollAndVotingScreen.kt

```diff
@@ -68,11 +79,26 @@ fun PollsAndVotingScreen(
         userRole = eventUIState.currentUserRole,
         onOptionVote = {
           eventViewModel.castPollVoteAtomically(
-              pollId = selectedPollId, optionIndex = it, { displayViewPollDialog = true })
+              pollId = selectedPollId,
+              optionIndex = it,
+              onSuccess = { displayViewPollDialog = true },
+              onFailure = {
+                Toast.makeText(
+                        context, context.getString(R.string.polls_vote_failure), Toast.LENGTH_SHORT)
+                    .show()
+              })
         },
         onPollDelete = {
           eventViewModel.deletePollAtomically(
-              pollId = it, onSuccess = { displayVotePollDialog = false })
+              pollId = it,
+              onSuccess = { displayVotePollDialog = false },
+              onFailure = {
+                Toast.makeText(
+                        context,

```

Comment:
You could have created a global showToast method, in order to orevent queueing the toasts